src/backend/commands/explain_gp.c

```diff
@@ -1545,6 +1583,131 @@ nodeSupportWorkfileCaching(PlanState *planstate)
 }
 
+static void
+cdbexplain_NodeSummary(ExplainState *es, CdbExplain_NodeSummary *ns) {
+	int			i;
+	char		aggbuf[100];
+	
+	if (es->format == EXPLAIN_FORMAT_TEXT)
+	{
+		/*
+			* create a header for all stats: separate each individual stat by an
+			* underscore, separate the grouped stats for each node by a slash
+			*/
+		appendStringInfoSpaces(es->str, es->indent * 2);
+		appendStringInfoString(es->str, "Node Summary:                  vmax       vsum       vcnt       imax\n");
+
+		int ns_spaces = es->indent * 2 + 2;
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->ntuples);
+		appendStringInfo(es->str, "ntuples:               %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->nloops);
+		appendStringInfo(es->str, "nloops:                %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->execmemused);
+		appendStringInfo(es->str, "execmemused:           %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->workmemused);
+		appendStringInfo(es->str, "workmemused:           %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->workmemwanted);
+		appendStringInfo(es->str, "workmemwanted:         %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->totalWorkfileCreated);
+		appendStringInfo(es->str, "totalWorkfileCreated:  %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		cdbexplain_formatAgg(aggbuf, sizeof(aggbuf), ns->totalPartTableScanned);
+		appendStringInfo(es->str, "totalPartTableScanned: %s\n", aggbuf);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		appendStringInfo(es->str, "segindex0: %d\n", ns->segindex0);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		appendStringInfo(es->str, "ninst: %d\n", ns->ninst);
+
+		appendStringInfoSpaces(es->str, ns_spaces);
+		appendStringInfoString(es->str, "StatInsts:\n");
+		appendStringInfoSpaces(es->str, ns_spaces + 2);
+		appendStringInfoString(es->str, "(segN) pstype starttime counter firsttuple startup total ntuples ntuples2 nloops nfiltered1 nfiltered2 execmemused workmemused workmemwanted workfileCreated firststart numPartScanned bnotes enotes, nworkers_launched\n");
+	}
+	else {
+		ExplainOpenGroup("CdbExplain_NodeSummary", "CdbExplain_NodeSummary", true, es);
+
+		ExplainPropertyAgg("ntuples", ns->ntuples, es);
+		ExplainPropertyAgg("nloops", ns->nloops, es);
+		ExplainPropertyAgg("execmemused", ns->execmemused, es);
+		ExplainPropertyAgg("workmemused", ns->workmemused, es);
+		ExplainPropertyAgg("workmemwanted", ns->workmemwanted, es);
+		ExplainPropertyAgg("totalWorkfileCreated", ns->totalWorkfileCreated, es);
+		ExplainPropertyAgg("totalPartTableScanned", ns->totalPartTableScanned, es);
```

Comment:
Add some tests for these codes which is not text format.

Reply:
I added test for non text format